dev/TeachingTip/TeachingTip.cpp

```diff
@@ -1235,6 +1278,21 @@ winrt::TeachingTipPlacementMode TeachingTip::DetermineEffectivePlacement()
             bool leftCenterAvailable = true;
             bool leftTopAvailable = true;
             bool leftBottomAvailable = true;
+            auto screenSize = winrt::Size({ 0.0f, 0.0f });
+            if (!shouldConstrainToRootBounds)
+            {
+                if (m_useTestScreenBounds)
+                {
+                    screenSize = winrt::Size(m_testScreenBounds.Width, m_testScreenBounds.Height);
+                }
+                else
+                {
+                    auto displayInfo = winrt::DisplayInformation::GetForCurrentView();
+                    auto rawPixelsPerViewPixel = displayInfo.RawPixelsPerViewPixel();
+                    screenSize = winrt::Size(displayInfo.ScreenHeightInRawPixels() / static_cast<float>(rawPixelsPerViewPixel),
+                                             displayInfo.ScreenWidthInRawPixels() / static_cast<float>(rawPixelsPerViewPixel));
+                }
+            }
 
             auto windowBounds = m_useTestWindowBounds ? m_testWindowBounds : winrt::Window::Current().CoreWindow().Bounds();
             auto targetBounds = m_currentTargetBounds;
```

Comment:
Should we be using XamlRoot instead of CoreWindow Bounds?

Reply:
XamlRoot doesn't have X and Y information which we need for proper placement.

---
In reply to: [268776564](https://github.com/Microsoft/microsoft-ui-xaml/pull/480#discussion_r268776564) [](ancestors = 268776564)